src/firefly/java/edu/caltech/ipac/firefly/server/db/BaseDbAdapter.java

```diff
@@ -0,0 +1,166 @@
+
+import static edu.caltech.ipac.firefly.data.TableServerRequest.INCL_COLUMNS;
+
+/**
+ * @author loi
+ * @version $Id: DbInstance.java,v 1.3 2012/03/15 20:35:40 loi Exp $
+ */
+abstract public class BaseDbAdapter implements DbAdapter {
+
+    private static final String DD_INSERT_SQL = "insert into dd values (?,?,?,?,?,?,?,?,?,?)";
+    private static final String DD_CREATE_SQL = "create table if not exists dd "+
+            "(" +
+            "  cname    varchar(1023)" +
+            ", label    varchar(1023)" +
+            ", type     varchar(1023)" +
+            ", units    varchar(1023)" +
+            ", format   varchar(1023)" +
+            ", width    int" +
+            ", visibility varchar(1023)" +
+            ", sortable boolean" +
+            ", filterable boolean" +
+            ", desc     varchar(1023)" +
+            ")";
+
+    private static final String META_INSERT_SQL = "insert into meta values (?,?)";
+    private static final String META_CREATE_SQL = "create table if not exists meta "+
+            "(" +
+            "  key      varchar(1023)" +
+            ", value    varchar(2023)" +
+            ")";
+
+
+    public String createMetaSql(DataType[] dataDefinitions) {
+        return META_CREATE_SQL;
+    }
+
+    public String insertMetaSql(DataType[] dataDefinitions) {
+        return META_INSERT_SQL;
+    }
+
+    public String createDDSql(DataType[] dataDefinitions) {
+        return DD_CREATE_SQL;
+    }
+
+    public String insertDDSql(DataType[] dataDefinitions) {
+        return DD_INSERT_SQL;
+    }
+
+    public String createDataSql(DataType[] dtTypes, String tblName) {
+        tblName = StringUtils.isEmpty(tblName) ? "data" : tblName;
+        List<String> coldefs = new ArrayList<>();
+        for(DataType dt : dtTypes) {
+            coldefs.add( dt.getKeyName() + " " + getDataType(dt.getDataType()));
+        }
+
+        return String.format("create table if not exists %s (%s)", tblName, StringUtils.toString(coldefs, ","));
+    }
+
+    public String insertDataSql(DataType[] dtTypes, String tblName) {
+        tblName = StringUtils.isEmpty(tblName) ? "data" : tblName;
+
+        String[] var = new String[dtTypes.length];
+        Arrays.fill(var , "?");
+        return String.format("insert into %s values(%s)", tblName, StringUtils.toString(var, ","));
+    }
+
+    public String getMetaSql() {
+        return "select * from meta";
+    }
+
+    public String getDDSql() {
+        return "select * from dd";
+    }
+
+    public String selectPart(TableServerRequest treq) {
+        String cols = treq.getParam(INCL_COLUMNS);
+        cols = "select " + (StringUtils.isEmpty(cols) ? "*" : cols);
+        return cols;
+    }
+
+    public String fromPart(TableServerRequest treq) {
+        String from = TableDbUtil.getDatasetID(treq);
+        from = StringUtils.isEmpty(from) ? treq.getParam(TableServerRequest.SQL_FROM) : from;
+        from = "from " + (StringUtils.isEmpty(from) ? "data" : from);
```

Comment:
May we need a `StringUtils.get(String str, String default)`
```
from = "from " +  StringUtils.get(from,"data")
```

Reply:
Sure, I can add that.